administrator/components/com_content/controllers/article.php

```diff
@@ -161,29 +161,24 @@ protected function postSaveHook(JModelLegacy $model, $validData)
 		$itemModel = JModelAdmin::getInstance('Item', 'MenusModel');
 		$itemModel->addTablePath(JPATH_ADMINISTRATOR . '/components/com_menus/tables');
 
-		$componentId = JComponentHelper::getComponent('com_content')->id;
-
-		JLoader::register('ContentHelper', JPATH_ADMINISTRATOR . '/components/com_content/helpers/content.php');
-		$articleId = ContentHelper::getarticleid($validData['title']);
-
-		$vaData['id'] = 0;
-		$vaData['menutype'] = $validData['menulink']['menutype'];
-		$vaData['title'] = $validData['menulink']['menutitle'];
-		$vaData['alias'] = $validData['menulink']['menualias'];
-		$vaData['link'] = 'index.php?option=com_content&view=article&id=' . $articleId;
-		$vaData['type'] = 'component';
-		$vaData['published'] = 1;
-		$vaData['parent_id'] = 1;
-		$vaData['level'] = 1;
-		$vaData['component_id'] = $componentId;
-		$vaData['browserNav'] = 0;
-		$vaData['access'] = 1;
-		$vaData['template_style_id'] = 0;
-		$vaData['home'] = 0;
-		$vaData['language'] = '*';
-		$vaData['client_id'] = 0;
-
-		$itemModel->save($vaData);
+		$menuData['id'] = 0;
+		$menuData['menutype'] = $validData['menulink']['menutype'];
+		$menuData['title'] = $validData['menulink']['menutitle'];
+		$menuData['alias'] = $validData['menulink']['menualias'];
+		$menuData['link'] = 'index.php?option=com_content&view=article&id=' . $model->getState($this->context . '.id');
```

Comment:
Here is you will create variable then you will get more readable code. :wink: So, I would say it depends where you are using. It is good if you create variable for `$articleId`.
See which one is more readable.
```php
// More readable
$menuData['link'] = 'index.php?option=com_content&view=article&id=' . $articleId;

// Complex to read
$menuData['link'] = 'index.php?option=com_content&view=article&id=' . $model->getState($this->context . '.id');
``